Nodejs/Product/Analysis/Analysis/AnalysisLimits.cs

```diff
@@ -34,6 +34,17 @@ internal class AnalysisLimits {
             MaxObjectLiteralProperties = 50;
             MaxObjectKeysTypes = 5;
             MaxMergeTypes = 5;
+            NestedModulesLimit = AnalysisConstants.MaxAnalysisDepthQualty;
```

Comment:
For the sake of readability and consistency, let's not add this to AnalysisConstants, and instead just include the number. The comment about reasoning is very helpful, though, and should be included where the NestedModulesLimit property is defined.